@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -449,38 +450,15 @@ function placePointLabelChoosingAnchor(
     // Placements options will be read from label.layoutStyle.placements in final solution.
     const placements = placementCentered ? anchorPlacementsCentered : anchorPlacementsCornered;
     const placementsNum = placements.length;
-    // Find current anchor placement on the optional placements list or best matching position.
+    // Find current anchor placement on the optional placements list.
     // Index of exact match.
-    let matchIdx = -1;
-    // Index of closest placement to the base one.
-    let closeIdxOff = 0;
-    let closeDiff = Infinity;
-    for (let i = 0; i < placementsNum; ++i) {
-        const placement = placements[i];
-        const hDiff = placement.h - basePlacement.h;
-        const vDiff = placement.v - basePlacement.v;
-        const manhattanDist = Math.abs(hDiff) + Math.abs(vDiff);
-        if (manhattanDist === 0) {
-            matchIdx = i;
-            closeIdxOff = 0;
-            break;
-        } else if (manhattanDist < closeDiff) {
-            closeDiff = manhattanDist;
-            closeIdxOff = i;
-        }
-    }
-
+    const matchIdx = placements.findIndex(p => p.h === basePlacement.h && p.v === basePlacement.v);
+    assert(matchIdx >= 0);
     // Will be true if all text placements are invisible.
     let allInvisible: boolean = true;
-    // If current anchor found on the list we start from it.
-    // If not listed in placements list (matchIdx === -1) we start from current anchor setup, but
-    // then immediately jump to the best matching option and iterate the over rest of placements.
-    // The number of iterations varies from `placementsNum` if perfect match found to
-    // `placementsNum + 1` otherwise (current placement + all optional).
-    const triesNum = matchIdx < 0 ? placementsNum + 1 : placementsNum;
-    for (let i = matchIdx, tryIdx = 0; tryIdx < triesNum; ++i, ++tryIdx) {
-        const anchorPlacement =
-            i < 0 ? basePlacement : placements[(i + closeIdxOff) % placementsNum];
+    // Iterate all placements starting from current one.
+    for (let i = matchIdx, tryIdx = 0; tryIdx < placementsNum; ++i, ++tryIdx) {
```

Comment:
minor: tryIdx is not needed. If you want to know if it's the base placement you can compare i === matchIdx.
You can fix this later.

Reply:
Of course, but it is basically the same, `tryIdx` is not here for testing the base (first) run, but for counting the number of tests, so we are sure that we made a full round (circular iteration).